mlir/lib/Analysis/DataFlow/IntegerRangeAnalysis.cpp

```diff
@@ -155,9 +155,9 @@ void IntegerRangeAnalysis::visitNonControlFlowArguments(
                                   Type boundType, bool getUpper) {
     unsigned int width = ConstantIntRanges::getStorageBitwidth(boundType);
     if (loopBound.has_value()) {
-      if (loopBound->is<Attribute>()) {
+      if (isa<Attribute>(*loopBound)) {
```

Comment:
dyn_cast?

Reply:
Thank you for pointing this out!  Fixed.